source-code/core2/src/fullExample.ts

```diff
@@ -0,0 +1,105 @@
+import { createApp } from "./app/createApp.js"
+import type { InlangConfig } from "./config/schema.js"
+import type {  MessageLintRule } from "./lint/api.js"
+import type { Plugin_Proposal_2 } from "./plugin/api.js"
+
+// --------------------- LINT RULE ---------------------
+
+const missingMessage: MessageLintRule = {
+	id: "inlang.missingMessage",
+	displayName: {
```

Comment:
we probably need a "fallback" prop, because "en" won't be present everywhere

Reply:
We should force `en` to be available.

Reply:
No, If I only create a rule for my italian company and we only support it and de, why should I add an english text?

Reply:
> No, If I only create a rule for my italian company and we only support it and de, why should I add an english text? 

Because inlang apps (like the editor) are globalized. 

The fallback is English. If you want to add translations for the display name of your lint rule, go for it and add Italian. Really minor issue. Forcing English to be provided seems vastly superior to a fallback option in case a "company doesn't want to write two words" for the English display name. I will push against the proposal.


<img width="431" alt="CleanShot 2023-07-21 at 12 26 28@2x" src="https://github.com/inlang/inlang/assets/35429197/13cd7749-53ba-4f69-92c4-61b7ed3aa871">

Reply:
If I write a private plugin (no plan to publish it) ,why should I care about en?
It is not just the display name.. also the lint messages would need to be translated accordingly

Reply:
> If I write a private plugin (no plan to publish it) ,why should I care about en?

Because translators in your company might use the editor in English. 

> It is not just the display name.. also the lint messages would need to be translated accordingly

Correct. Forgot to add it. 

TBH it seems very straightforward that lint messages must provide `en` for the sake of providing a good UX across the inlang ecosystem.

Reply:
I would say 98% of our repos are using `en` as their default. However https://github.com/tournesol-app/tournesol/pull/1610 for example have `fr` as a default.

Reply:
This is about lint rules. I am 200% certain that we can force everyone to provide `en` for the sake of being able to globalize inlang apps themselves.

Reply:
I have built a lot of web projects in the past and can confidently say that NOT all webpages want English by default. Especially for the EU market, people often want their language as the default. (Speaking about German, French, Italian, Spanish, Portuguese, ...). 

Proposal: We can derive the required languages for lint translations from `config.languageTags` via TS?

Reply:
GUYS OMG STOP MIXING STUFF. 

THIS IS ABOUT LINT RULES, NOT ABOUT ORGANIZATIONS USING INLANG TO GLOBALIZE THEIR APPS. PR will follow but sorry this discussion is completely derailed.

Reply:
Sorry! I thought my input would be valuable.
GitHub didn't update my feed so I didn't saw your comment: 

> This is about lint rules. I am 200% certain that we can force everyone to provide en for the sake of being able to globalize inlang apps themselves.

Reply:
PR is up https://github.com/inlang/inlang/pull/1155. Closing this in favor of the PR